include/capmap-mappings.h

```diff
@@ -93,7 +93,65 @@ class LoadMap : public Map {
   SparseRange ranges_;
 };
 
-// TODO: Provide other `Map` derivatives.
+// Ranges with given permissions
+class BasicMap : public Map {
```

Comment:
> a dozen different classes that only differ in values

I agree that that's undesirable, but I also think it won't be that common! I'd suggest that a `SimplePermissionMap` or something could serve that purpose where it makes sense (especially for user permissions and non-capability loads and stores), but most permissions require some kind of special `try_combine()` handling, and that's the only non-trivial function required by the `Map` abstract class.

> Tracking Store permissions seems like a desirable thing to do, but the code will be almost exactly the same.

Right, but we can implement that using some (runtime or template) parameter for `LoadMap` specifically, or perhaps one day we could reimplement `LoadMap` in terms of `SimplePermissionMap`. We could then do something similar for `LoadCapMap`/`StoreCapMap`/`StoreLocalCapMap`/`LoadMutableCapMap`, which all share alignment behaviours.

In any case, if you want this for your experiments, then let's keep it (but ideally with a different name, as noted).


As an aside: we're already paying a runtime performance cost by having the virtual function call. That was a cost that I thought we could pay for the flexibility it provides, but if we end up moving a lot of logic to run-time checks anyway then we should reconsider that, to avoid paying the cost twice. I don't want to do that _yet_, because it's far too early to make such optimisation decisions, but I do think it likely that we'll refactor a lot of this code later.

Reply:
Sorry it's taken so long to get back on everything -- I was out sick for almost a week and also have had conference prep to do.

I've renamed it to PermissionMap (commit [ce2eb51](https://github.com/capablevms/compartment-mapper/pull/3/commits/ce2eb51ca1615a90b6c9ff51ea9a5e81ef92b095))

I'm not sure if there were any other explicit changes you were requesting. As you say, this could be obsoleted by a more sophisticated kind of map, but I'm not sure I should be trying to add that halfway through a pull request. Is it okay as it stands?

Reply:
It's Ok, in the spirit of experimentation. I think we'll want to refactor these later, that's all.